### PR TITLE
core: Update default sampler to new DatadogSampler

### PR DESCRIPTION
**Pull request by brettlangdon**
We have made a change to ensure that, if no explicitly defined rules match and no global sample rate is defined, we continue to use a `RateByServiceSampler`.

I tried to unify the logic here a bit more to make it easier to drop in a `DatadogSampler` as the default without needing to rely on configuring/passing priority samplers around, hopefully it makes it more clear.

How people can utilize this new sampler:

1) Set `DD_TRACE_SAMPLE_RATE=` env variable to set a global rate limit
  a) this will by-pass the `RateByServiceSampler`
2) Configure and pass in the sampler manually:

```python
from ddtrace import tracer
from ddtrace.sampler import DatadogSampler

tracer.configure(sampler=DatadogSampler(sample_rate=1.0, rules=[ ... ], rate_limit=100))
```

### CODE REVIEW

File: ddtrace/sampler.py [hunk 1]
```diff
@@ -175,11 +200,15 @@ def sample(self, span):
...
-            # No rule matches, use the default sampler
+            # If no rules match, use our defualt sampler
+            # This may be a RateByServiceSampler
             matching_rule = self.default_sampler
 
         # Sample with the matching sampling rule
-        span.set_metric(SAMPLING_RULE_DECISION, matching_rule.sample_rate)
+        # Only set if it isn't the default RateByServiceSampler
+        #   since that gets it's own metric
+        if isinstance(matching_rule, SamplingRule):
```
Comment (furmmon) [inline ddtrace/sampler.py]: Rate limiter is only meant to be applied to Rules, as setting 100% of rate might generate uncontrolled traffic.
Rate by service gives a number of trace/s controlled in the agent, this is controlled and always under a fix number/s

File: ddtrace/sampler.py [hunk 2]
```diff
@@ -109,15 +115,15 @@ def set_sample_rate_by_service(self, rate_by_service):
...
+class DatadogSampler(BaseSampler, BasePrioritySampler):
     """
     This sampler is currently in ALPHA and it's API may change at any time, use at your own risk.
     """
     __slots__ = ('default_sampler', 'limiter', 'rules')
 
     NO_RATE_LIMIT = -1
-    DEFAULT_RATE_LIMIT = 100
-    DEFAULT_SAMPLE_RATE = 1.0
+    DEFAULT_RATE_LIMIT = NO_RATE_LIMIT
```
Comment (furmmon) [inline ddtrace/sampler.py]: Rate limiter should always be at 100 by default

File: ddtrace/sampler.py (hunk 1 above)
Comment (brettlangdon) [reply ddtrace/sampler.py]: updated

File: ddtrace/sampler.py [hunk 3]
```diff
@@ -109,30 +115,40 @@ def set_sample_rate_by_service(self, rate_by_service):
...
+    DEFAULT_SAMPLE_RATE = None
 
     def __init__(self, rules=None, default_sample_rate=None, rate_limit=None):
         """
         Constructor for DatadogSampler sampler
 
         :param rules: List of :class:`SamplingRule` rules to apply to the root span of every trace, default no rules
         :type rules: :obj:`list` of :class:`SamplingRule`
-        :param default_sample_rate: The default sample rate to apply if no rules matched (default: 1.0)
+        :param default_sample_rate: The default sample rate to apply if no rules matched (default: ``1.0``)
```
Comment (furmmon) [inline ddtrace/sampler.py]: The default is wrong as it's disabled by default

Comment (brettlangdon) [reply ddtrace/sampler.py]:
```suggestion
        :param default_sample_rate: The default sample rate to apply if no rules matched (default: ``None``/Use :class:`RateByServiceSampler` only)
```

**Pull request merged**